addons/website_sale/models/sale_order.py

```diff
@@ -292,121 +276,69 @@ def _update_address(self, partner_id, fnames=None):
             # Only add the main partner as follower of the order
             self._message_subscribe([partner_id])
 
-    def _cart_update(self, product_id, line_id=None, add_qty=0, set_qty=0, **kwargs):
-        """ Add or set product quantity, add_qty can be negative """
+    def _cart_add(self, product_id:int, quantity:int|float=1.0, **kwargs):
```

Comment:
Since we now load the record every time before calling `_cart_add`, why only giving the id?